gobblin-utility/src/main/java/gobblin/util/SchedulerUtils.java

```diff
@@ -178,6 +215,96 @@ public boolean accept(File file) {
   }
 
   /**
+   * Recursively load job configuration files under given URI of **directory of config files folder**
+   * todo
+   */
+  private static void loadGenericJobConfigsRecursive(List<Properties> jobConfigs, Properties rootProps,
+      Set<String> jobConfigFileExtensions, URI uri)
+      throws ConfigurationException, IOException {
+
+    /* create the generic file system */
+    org.apache.hadoop.conf.Configuration conf = new org.apache.hadoop.conf.Configuration();
+    FileSystem filesystem = FileSystem.get(uri, conf);
```

Comment:
If the last argument of the method signature is of type `Path`, you can instead do `path.getFileSystem(conf)`.